# Data Cleaning

In [1]:
import pandas as pd

### 1.1 Loading the data into dataframes

In [2]:
companies=pd.read_csv('C:/Users/Administrator/Desktop/pandas/Teclov/companies.txt',encoding='ansi',delimiter='\t')
companies

permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
...                                                  ...   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name  \
0                                              #fame   
1                                           :Qounter   
2                             (THE) ONE of THEM,Inc.   
3                                            0-6.com   
4                                   004 Technologies   
...                                              ...   
66363                  ZZNode Science and Technology   
66364                          Zzzzapp Wireless ltd.   
66365                                         ÁERON   
66366                                         Ôasys   
66367  İnovatiff Reklam ve Tanıtım Hizmetleri Tic   

                            homepage_url  \
0                     http://livfame.com   
1                 http://www.qounter.com   
2                    http://oneofthem.jp   
3                     http://www.0-6.com   
4      http://004gmbh.de/en/004-interact   
...                                  ...   
66363              http://www.zznode.com   
66364             http://www.zzzzapp.com   
66365               http://www.aeron.hu/   
66366               http://www.oasys.io/   
66367               http://inovatiff.com   

                                           category_list     status  \
0                                                  Media  operating   
1      Application Platforms|Real Time|Social Network...  operating   
2                                      Apps|Games|Mobile  operating   
3                                            Curated Web  operating   
4                                               Software  operating   
...                                                  ...        ...   
66363                                Enterprise Software  operating   
66364        Advertising|Mobile|Web Development|Wireless  operating   
66365                                                NaN  operating   
66366  Consumer Electronics|Internet of Things|Teleco...  operating   
66367                 Consumer Goods|E-Commerce|Internet  operating   

      country_code state_code                 region           city  \
0              IND         16                 Mumbai         Mumbai   
1              USA         DE             DE - Other  Delaware City   
2              NaN        NaN                    NaN            NaN   
3              CHN         22                Beijing        Beijing   
4              USA         IL  Springfield, Illinois      Champaign   
...            ...        ...                    ...            ...   
66363          CHN         22                Beijing        Beijing   
66364          HRV         15                  Split          Split   
66365          NaN        NaN                    NaN            NaN   
66366          USA         CA            SF Bay Area  San Francisco   
66367          NaN        NaN                    NaN            NaN   

       founded_at  
0             NaN  
1      04-09-2014  
2             NaN  
3      01-01-2007  
4      01-01-2010  
...           ...  
66363         NaN  
66364  13-05-2012  
66365  01-01-2011  
66366  01-01-2014  
66367         NaN  

[66368 rows x 10 columns]

In [3]:
rounds2=pd.read_csv('C:/Users/Administrator/Desktop/pandas/Teclov/rounds2.csv',encoding='ansi')
rounds2

company_permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
...                                                   ...   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
...                                                 ...                ...   
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  
0                       B  05-01-2015         10000000.0  
1                       A  14-10-2014                NaN  
2                     NaN  01-03-2014           700000.0  
3                       B  30-01-2014          3406878.0  
4                       A  19-03-2008          2000000.0  
...                   ...         ...                ...  
114944                NaN  01-03-2014            41313.0  
114945                NaN  01-05-2013            32842.0  
114946                  A  01-08-2014                NaN  
114947                NaN  01-01-2015            18192.0  
114948                NaN  01-10-2013            14851.0  

[114949 rows x 6 columns]

### 1.2 Unique companies present in rounds2 and companies

### For companies

In [4]:
companyname=companies['permalink'].apply(lambda x:x.replace('/Organization/',''))
companyname=companyname.str.lower()


In [5]:
companies['Company_Name']=companyname

In [6]:
companies['Company_Name'].nunique()

66368

In [7]:
companies

permalink  \
0                                    /Organization/-Fame   
1                                 /Organization/-Qounter   
2                    /Organization/-The-One-Of-Them-Inc-   
3                                  /Organization/0-6-Com   
4                         /Organization/004-Technologies   
...                                                  ...   
66363  /Organization/Zznode-Science-And-Technology-Co...   
66364                          /Organization/Zzzzapp-Com   
66365                               /Organization/ÁEron   
66366                             /Organization/ÔAsys-2   
66367  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                name  \
0                                              #fame   
1                                           :Qounter   
2                             (THE) ONE of THEM,Inc.   
3                                            0-6.com   
4                                   004 Technologies   
...                                              ...   
66363                  ZZNode Science and Technology   
66364                          Zzzzapp Wireless ltd.   
66365                                         ÁERON   
66366                                         Ôasys   
66367  İnovatiff Reklam ve Tanıtım Hizmetleri Tic   

                            homepage_url  \
0                     http://livfame.com   
1                 http://www.qounter.com   
2                    http://oneofthem.jp   
3                     http://www.0-6.com   
4      http://004gmbh.de/en/004-interact   
...                                  ...   
66363              http://www.zznode.com   
66364             http://www.zzzzapp.com   
66365               http://www.aeron.hu/   
66366               http://www.oasys.io/   
66367               http://inovatiff.com   

                                           category_list     status  \
0                                                  Media  operating   
1      Application Platforms|Real Time|Social Network...  operating   
2                                      Apps|Games|Mobile  operating   
3                                            Curated Web  operating   
4                                               Software  operating   
...                                                  ...        ...   
66363                                Enterprise Software  operating   
66364        Advertising|Mobile|Web Development|Wireless  operating   
66365                                                NaN  operating   
66366  Consumer Electronics|Internet of Things|Teleco...  operating   
66367                 Consumer Goods|E-Commerce|Internet  operating   

      country_code state_code                 region           city  \
0              IND         16                 Mumbai         Mumbai   
1              USA         DE             DE - Other  Delaware City   
2              NaN        NaN                    NaN            NaN   
3              CHN         22                Beijing        Beijing   
4              USA         IL  Springfield, Illinois      Champaign   
...            ...        ...                    ...            ...   
66363          CHN         22                Beijing        Beijing   
66364          HRV         15                  Split          Split   
66365          NaN        NaN                    NaN            NaN   
66366          USA         CA            SF Bay Area  San Francisco   
66367          NaN        NaN                    NaN            NaN   

       founded_at                                   Company_Name  
0             NaN                                          -fame  
1      04-09-2014                                       -qounter  
2             NaN                          -the-one-of-them-inc-  
3      01-01-2007                                        0-6-com  
4      01-01-2010                               004-technologies  
...           ..

### For rounds2

In [8]:
sr=rounds2['company_permalink'].apply(lambda x:x.replace('/organization/',''))
company_name=sr.apply(lambda x:x.replace('/ORGANIZATION/',''))
company_name=company_name.str.lower()


In [9]:
rounds2['Company_Name']=company_name

In [10]:
rounds2['Company_Name'].nunique()

66368

In [11]:
rounds2

company_permalink  \
0                                     /organization/-fame   
1                                  /ORGANIZATION/-QOUNTER   
2                                  /organization/-qounter   
3                     /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                                   /organization/0-6-com   
...                                                   ...   
114944                          /organization/zzzzapp-com   
114945                          /ORGANIZATION/ZZZZAPP-COM   
114946                               /organization/ãeron   
114947                             /ORGANIZATION/ÔASYS-2   
114948  /organization/ä°novatiff-reklam-ve-tanä±tä±m-h...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
...                                                 ...                ...   
114944  /funding-round/8f6d25b8ee4199e586484d817bceda05   convertible_note   
114945  /funding-round/ff1aa06ed5da186c84f101549035d4ae               seed   
114946  /funding-round/59f4dce44723b794f21ded3daed6e4fe            venture   
114947  /funding-round/35f09d0794651719b02bbfd859ba9ff5               seed   
114948  /funding-round/af942869878d2cd788ef5189b435ebc4              grant   

       funding_round_code   funded_at  raised_amount_usd  \
0                       B  05-01-2015         10000000.0   
1                       A  14-10-2014                NaN   
2                     NaN  01-03-2014           700000.0   
3                       B  30-01-2014          3406878.0   
4                       A  19-03-2008          2000000.0   
...                   ...         ...                ...   
114944                NaN  01-03-2014            41313.0   
114945                NaN  01-05-2013            32842.0   
114946                  A  01-08-2014                NaN   
114947                NaN  01-01-2015            18192.0   
114948                NaN  01-10-2013            14851.0   

                                         Company_Name  
0                                               -fame  
1                                            -qounter  
2                                            -qounter  
3                               -the-one-of-them-inc-  
4                                             0-6-com  
...                                               ...  
114944                                    zzzzapp-com  
114945                                    zzzzapp-com  
114946                                         ãeron  
114947                                       ã”asys-2  
114948  ä°novatiff-reklam-ve-tanä±tä±m-hizmetleri-tic  

[114949 rows x 7 columns]

In [12]:
rounds2['Company_Name'][~rounds2['Company_Name'].isin(companies['Company_Name'])].count()

0

### Merging the two data frames

In [13]:
master_frame=pd.merge(companies,rounds2,how='inner')
master_frame

permalink  \
0                                     /Organization/-Fame   
1                                  /Organization/-Qounter   
2                                  /Organization/-Qounter   
3                     /Organization/-The-One-Of-Them-Inc-   
4                                   /Organization/0-6-Com   
...                                                   ...   
114944                          /Organization/Zzzzapp-Com   
114945                          /Organization/Zzzzapp-Com   
114946                               /Organization/ÁEron   
114947                             /Organization/ÔAsys-2   
114948  /Organization/Ä°Novatiff-Reklam-Ve-Tanä±Tä±M-H...   

                                                 name            homepage_url  \
0                                               #fame      http://livfame.com   
1                                            :Qounter  http://www.qounter.com   
2                                            :Qounter  http://www.qounter.com   
3                              (THE) ONE of THEM,Inc.     http://oneofthem.jp   
4                                             0-6.com      http://www.0-6.com   
...                                               ...                     ...   
114944                          Zzzzapp Wireless ltd.  http://www.zzzzapp.com   
114945                          Zzzzapp Wireless ltd.  http://www.zzzzapp.com   
114946                                         ÁERON    http://www.aeron.hu/   
114947                                         Ôasys    http://www.oasys.io/   
114948  İnovatiff Reklam ve Tanıtım Hizmetleri Tic    http://inovatiff.com   

                                            category_list     status  \
0                                                   Media  operating   
1       Application Platforms|Real Time|Social Network...  operating   
2       Application Platforms|Real Time|Social Network...  operating   
3                                       Apps|Games|Mobile  operating   
4                                             Curated Web  operating   
...                                                   ...        ...   
114944        Advertising|Mobile|Web Development|Wireless  operating   
114945        Advertising|Mobile|Web Development|Wireless  operating   
114946                                                NaN  operating   
114947  Consumer Electronics|Internet of Things|Teleco...  operating   
114948                 Consumer Goods|E-Commerce|Internet  operating   

       country_code state_code       region           city  founded_at  \
0               IND         16       Mumbai         Mumbai         NaN   
1               USA         DE   DE - Other  Delaware City  04-09-2014   
2               USA         DE   DE - Other  Delaware City  04-09-2014   
3               NaN        NaN          NaN            NaN         NaN   
4               CHN         22      Beijing        Beijing  01-01-2007   
...             ...        ...          ...            ...         ...   
114944          HRV         15        Split          Split  13-05-2012   
114945          HRV         15        Split          Split  13-05-2012   
114946          NaN        NaN          NaN            NaN  01-01-2011   
114947          USA         CA  SF Bay Area  San Francisco  01-01-2014   
114948          NaN        NaN          NaN            NaN         NaN   

                                         Company_Name  \
0                                               -fame   
1                                            -qounter   
2                                            -qounter   
3                               -the-one-of-them-inc-   
4                                             0-6-com   
...                                               ...   
114944                                    zzzzapp-com   
114945                                    zzzzapp-com   
114946                                         ãeron   
114947            

In [14]:
master_frame.shape

(114949, 17)

# Problem 2: Funding Type Analysis 

## Part-1

In [15]:
import numpy as np
from scipy import stats
a=master_frame[(master_frame['funding_round_type']=='angel')|(master_frame['funding_round_type']=='seed')|(master_frame['funding_round_type']=='venture')|(master_frame['funding_round_type']=='private_equity')]
a.groupby('funding_round_type')['raised_amount_usd'].agg({'funding_round_type':stats.mode})

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:4: FutureWarning: using a dict on a Series for aggregation
is deprecated and will be removed in a future version. Use                 named aggregation instead.

    >>> grouper.agg(name_1=func_1, name_2=func_2)

  after removing the cwd from sys.path.


funding_round_type
funding_round_type                        
angel                  ([500000.0], [276])
private_equity       ([100000000.0], [67])
seed                 ([1000000.0], [1065])
venture             ([10000000.0], [1809])


### VENTURE investment type is most suitable for Teclov

## Part-2

In [16]:
invest=master_frame[(master_frame['raised_amount_usd']>5000000)&(master_frame['raised_amount_usd']<15000000)]
x=invest[(invest['funding_round_type']=='angel')|(invest['funding_round_type']=='seed')|(invest['funding_round_type']=='venture')|(invest['funding_round_type']=='private_equity')]
x.groupby('funding_round_type')['raised_amount_usd'].agg({'funding_round_type':stats.mode})
#venture is the most suitable invesment type

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: FutureWarning: using a dict on a Series for aggregation
is deprecated and will be removed in a future version. Use                 named aggregation instead.

    >>> grouper.agg(name_1=func_1, name_2=func_2)

  This is separate from the ipykernel package so we can avoid doing imports until


funding_round_type
funding_round_type                        
angel                  ([10000000.0], [9])
private_equity        ([10000000.0], [42])
seed                   ([6000000.0], [17])
venture             ([10000000.0], [1809])

# Problem-3 :Country Analysis

In [17]:
#top 9 countries receiving the highest total funding
invest=master_frame[(master_frame['raised_amount_usd']>5000000)&(master_frame['raised_amount_usd']<15000000)]
venture=invest[invest['funding_round_type']=='venture']
country=venture.groupby('country_code')['raised_amount_usd'].agg(np.sum).sort_values(ascending=False).head(9)
country

country_code
USA    8.960135e+10
GBR    4.941844e+09
CHN    3.861561e+09
CAN    3.192010e+09
ISR    2.250306e+09
IND    2.186544e+09
FRA    2.022425e+09
DEU    1.635026e+09
CHE    5.588196e+08
Name: raised_amount_usd, dtype: float64

In [18]:
top9=pd.DataFrame(country)
top9

raised_amount_usd
country_code                   
USA                8.960135e+10
GBR                4.941844e+09
CHN                3.861561e+09
CAN                3.192010e+09
ISR                2.250306e+09
IND                2.186544e+09
FRA                2.022425e+09
DEU                1.635026e+09
CHE                5.588196e+08

## Problem-4: Sector Analysis

In [19]:
sec=master_frame['category_list'].str.split('|').str[0]
sec

0                         Media
1         Application Platforms
2         Application Platforms
3                          Apps
4                   Curated Web
                  ...          
114944              Advertising
114945              Advertising
114946                      NaN
114947     Consumer Electronics
114948           Consumer Goods
Name: category_list, Length: 114949, dtype: object

In [20]:
mapping=pd.read_csv('C:/Users/Administrator/Desktop/pandas/Teclov/mapping.csv')
mapping

category_list  Automotive & Sports  Blanks  \
0                 NaN                    0       1   
1                  3D                    0       0   
2         3D Printing                    0       0   
3       3D Technology                    0       0   
4          Accounting                    0       0   
..                ...                  ...     ...   
683         Wholesale                    0       0   
684  Wine And Spirits                    0       0   
685          Wireless                    0       0   
686             Women                    0       0   
687      Young Adults                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
0                             0              0       0              0   
1                             0              0       0              1   
2                             0              0       0              1   
3                             0              0       0              1   
4                             0              0       0              0   
..                          ...            ...     ...            ...   
683                           0              0       0              0   
684                           0              1       0              0   
685                           0              1       0              0   
686                           0              0       0              0   
687                           0              0       0              0   

     News, Search and Messaging  Others  \
0                             0       0   
1                             0       0   
2                             0       0   
3                             0       0   
4                             0       0   
..                          ...     ...   
683                           0       1   
684                           0       0   
685                           0       0   
686                           0       1   
687                           0       1   

     Social, Finance, Analytics, Advertising  
0                                          0  
1                                          0  
2                                          0  
3                                          0  
4                                          1  
..                                       ...  
683                                        0  
684                                        0  
685                                        0  
686                                        0  
687                                        0  

[688 rows x 10 columns]

In [21]:
mapping.drop('Blanks',axis=1)

category_list  Automotive & Sports  Cleantech / Semiconductors  \
0                 NaN                    0                           0   
1                  3D                    0                           0   
2         3D Printing                    0                           0   
3       3D Technology                    0                           0   
4          Accounting                    0                           0   
..                ...                  ...                         ...   
683         Wholesale                    0                           0   
684  Wine And Spirits                    0                           0   
685          Wireless                    0                           0   
686             Women                    0                           0   
687      Young Adults                    0                           0   

     Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0                0       0              0                           0       0   
1                0       0              1                           0       0   
2                0       0              1                           0       0   
3                0       0              1                           0       0   
4                0       0              0                           0       0   
..             ...     ...            ...                         ...     ...   
683              0       0              0                           0       1   
684              1       0              0                           0       0   
685              1       0              0                           0       0   
686              0       0              0                           0       1   
687              0       0              0                           0       1   

     Social, Finance, Analytics, Advertising  
0                                          0  
1                                          0  
2                                          0  
3                                          0  
4                                          1  
..                                       ...  
683                                        0  
684                                        0  
685                                        0  
686                                        0  
687                                        0  

[688 rows x 9 columns]

In [22]:
mapping1=mapping.melt(id_vars='category_list',value_vars=['Automotive & Sports','Cleantech / Semiconductors','Entertainment','Health','Manufacturing','News, Search and Messaging','Others','Social, Finance, Analytics, Advertising'])
mapping_category=mapping1[mapping1['value']==1]
mapping_category.drop('value',axis=1)

category_list                                 variable
8      Adventure Travel                      Automotive & Sports
14            Aerospace                      Automotive & Sports
45                 Auto                      Automotive & Sports
46      Automated Kiosk                      Automotive & Sports
47           Automotive                      Automotive & Sports
...                 ...                                      ...
5433  Social Recruiting  Social, Finance, Analytics, Advertising
5434  Social Television  Social, Finance, Analytics, Advertising
5435      Social Travel  Social, Finance, Analytics, Advertising
5446    Stock Exchanges  Social, Finance, Analytics, Advertising
5479    Venture Capital  Social, Finance, Analytics, Advertising

[687 rows x 2 columns]

# Problem-5: Sector Analysis 2

### For country-USA

In [23]:
invest=master_frame[(master_frame['raised_amount_usd']>5000000)&(master_frame['raised_amount_usd']<15000000)]
con=invest[(invest['country_code']=='USA')&(invest['funding_round_type']=='venture')]
D1=pd.merge(con,mapping_category)
D1

permalink                     name  \
0         /Organization/1366-Technologies        1366 Technologies   
1         /Organization/1366-Technologies        1366 Technologies   
2         /Organization/1366-Technologies        1366 Technologies   
3               /Organization/908-Devices              908 Devices   
4               /Organization/908-Devices              908 Devices   
...                                   ...                      ...   
4991               /Organization/Vulcun-2                   VULCUN   
4992  /Organization/Walker-Company-Brands  Walker & Company Brands   
4993                 /Organization/Washio                   Washio   
4994                   /Organization/X-Ai                     x.ai   
4995                 /Organization/Xosoft                   XOSoft   

                     homepage_url       category_list     status country_code  \
0         http://www.1366tech.com       Manufacturing  operating          USA   
1         http://www.1366tech.com       Manufacturing  operating          USA   
2         http://www.1366tech.com       Manufacturing  operating          USA   
3           http://908devices.com       Manufacturing  operating          USA   
4           http://908devices.com       Manufacturing  operating          USA   
...                           ...                 ...        ...          ...   
4991        http://www.vulcun.com      Fantasy Sports  operating          USA   
4992  http://walkerandcompany.com              Beauty  operating          USA   
4993     http://www.getwashio.com  Service Industries  operating          USA   
4994                  http://x.ai                 B2B  operating          USA   
4995        http://www.xosoft.com       Data Security   acquired          USA   

     state_code         region           city  founded_at  \
0            MA         Boston        Bedford  01-01-2007   
1            MA         Boston        Bedford  01-01-2007   
2            MA         Boston        Bedford  01-01-2007   
3            MA         Boston         Boston  01-01-2012   
4            MA         Boston         Boston  01-01-2012   
...         ...            ...            ...         ...   
4991         CA    SF Bay Area  San Francisco  01-05-2011   
4992         CA    SF Bay Area      Palo Alto  18-04-2013   
4993         CA    Los Angeles    Los Angeles  01-01-2013   
4994         NY  New York City       New York  14-04-2014   
4995         MA         Boston     Burlington  01-01-1999   

               Company_Name                    company_permalink  \
0         1366-technologies      /organization/1366-technologies   
1         1366-technologies      /ORGANIZATION/1366-TECHNOLOGIES   
2         1366-technologies      /organization/1366-technologies   
3               908-devices            /ORGANIZATION/908-DEVICES   
4               908-devices            /ORGANIZATION/908-DEVICES   
...                     ...                                  ...   
4991               vulcun-2               /organization/vulcun-2   
4992  walker-company-brands  /ORGANIZATION/WALKER-COMPANY-BRANDS   
4993                 washio                 /ORGANIZATION/WASHIO   
4994                   x-ai                   /organization/x-ai   
4995                 xosoft                 /organization/xosoft   

                              funding_round_permalink funding_round_type  \
0     /funding-round/786f61aa9866f4471151285f5c56be36            venture   
1     /funding-round/82ace97530965cd2be8f262836b43ff5            venture   
2     /funding-round/ab99fc5a53717b1b53fd6aa5687c5fa9            venture   
3     /funding-round/6cf034e663f0c22047d4a82a499a685b            venture   
4     /funding-round/d0b6f0eb32eb1b13e776213580bccd7a            venture   
...                                               ...                ...   
4991  /funding-round/038c646ed6d1dda83e95caea4d8ad46e            venture   
4992  /funding-round/2a4440bac96b28cb2bdcb89bdb5a5e09       

In [24]:
tot_investment1=D1.groupby('variable')['raised_amount_usd'].agg(np.size).sort_values(ascending=False)
tot_investment1

variable
Others                                     1728.0
Cleantech / Semiconductors                 1494.0
Health                                      498.0
News, Search and Messaging                  449.0
Manufacturing                               353.0
Social, Finance, Analytics, Advertising     281.0
Entertainment                               158.0
Automotive & Sports                          35.0
Name: raised_amount_usd, dtype: float64

In [25]:
company_inv=D1.groupby(['variable','Company_Name'])['Company_Name'].agg(np.size).sort_values(ascending=False)
company_inv

variable                                 Company_Name         
Cleantech / Semiconductors               tigo-energy              7
Health                                   endogastric-solutions    6
Social, Finance, Analytics, Advertising  choicestream             6
Cleantech / Semiconductors               scienergy                6
News, Search and Messaging               sebacia                  5
                                                                 ..
Others                                   aplicor                  1
                                         anthology-solutions      1
                                         anodyne-health           1
                                         anchiva-systems          1
Automotive & Sports                      additech                 1
Name: Company_Name, Length: 3713, dtype: int64

In [26]:
tot_amount1=D1.groupby('variable').agg({'raised_amount_usd':np.sum}).sort_values(ascending=False,by='raised_amount_usd')
tot_amount1

raised_amount_usd
variable                                                  
Others                                        1.554731e+10
Cleantech / Semiconductors                    1.357599e+10
Health                                        4.490750e+09
News, Search and Messaging                    3.865548e+09
Manufacturing                                 3.218865e+09
Social, Finance, Analytics, Advertising       2.484532e+09
Entertainment                                 1.378120e+09
Automotive & Sports                           2.862028e+08

### For Country-GRB

In [27]:
con1=invest[(invest['country_code']=='GBR')&(invest['funding_round_type']=='venture')]
D2=pd.merge(con1,mapping_category)
D2

permalink  \
0                                /Organization/Abcodia   
1                          /Organization/Acacia-Pharma   
2                                /Organization/Apitope   
3                 /Organization/Aquapharm-Biodiscovery   
4                 /Organization/Aquapharm-Biodiscovery   
..                                                 ...   
313                               /Organization/Redeem   
314  /Organization/Sandwell-Community-Caring-Trust-...   
315                    /Organization/Valldata-Services   
316                         /Organization/Serentis-Ltd   
317                              /Organization/Spinvox   

                                       name                 homepage_url  \
0                                   Abcodia           http://abcodia.com   
1                             Acacia Pharma  http://www.acaciapharma.com   
2                                   Apitope          http://apitope.com/   
3                    Aquapharm Biodiscovery   http://www.aquapharm.co.uk   
4                    Aquapharm Biodiscovery   http://www.aquapharm.co.uk   
..                                      ...                          ...   
313                                  Redeem     http://www.redeem.co.uk/   
314  Sandwell Community Caring Trust (SCCT)    http://sandwellcct.org.uk   
315                       Valldata Services    http://www.valldata.co.uk   
316                                Serentis  http://serentis-pharma.com/   
317                                 SpinVox       http://www.spinvox.com   

                category_list     status country_code state_code       region  \
0               Biotechnology  operating          GBR         H9       London   
1               Biotechnology  operating          GBR         C3       London   
2               Biotechnology  operating          GBR         B7      Bristol   
3               Biotechnology     closed          GBR        NaN          NaN   
4               Biotechnology     closed          GBR        NaN          NaN   
..                        ...        ...          ...        ...          ...   
313  Environmental Innovation  operating          GBR        NaN          NaN   
314                Nonprofits  operating          GBR         A7   Birmingham   
315                Nonprofits  operating          GBR         P8  GBR - Other   
316           Pharmaceuticals  operating          GBR         C3       London   
317                 Messaging   acquired          GBR         B9       London   

           city  founded_at                          Company_Name  \
0        London  01-01-2010                               abcodia   
1     Cambridge  01-01-2006                         acacia-pharma   
2       Bristol  01-01-2002                               apitope   
3           NaN  01-01-2000                aquapharm-biodiscovery   
4           NaN  01-01-2000                aquapharm-biodiscovery   
..          ...         ...                                   ...   
313         NaN         NaN                                redeem   
314  Birmingham  01-01-1997  sandwell-community-caring-trust-scct   
315    Melksham         NaN                     valldata-services   
316   Cambridge         NaN                          serentis-ltd   
317      Marlow         NaN                               spinvox   

                                     company_permalink  \
0                                /organization/abcodia   
1                          /organization/acacia-pharma   
2                                /organization/apitope   
3                 /ORGANIZATION/AQUAPHARM-BIODISCOVERY   
4                 /organization/aquapharm-biodiscovery   
..                                                 ...   
313                               /ORGANIZATION/REDEEM   
314  /organization/sandwell-community-caring-trust-...   
315                    /ORGANIZATION/VALLDATA-SERVICES   
316                         /organization/serentis-lt

In [28]:
tot_investment2=D2.groupby('variable')['raised_amount_usd'].agg(np.size).sort_values(ascending=False)
tot_investment2

variable
Cleantech / Semiconductors                 104.0
Others                                      95.0
News, Search and Messaging                  30.0
Manufacturing                               27.0
Entertainment                               27.0
Social, Finance, Analytics, Advertising     20.0
Health                                      12.0
Automotive & Sports                          3.0
Name: raised_amount_usd, dtype: float64

In [29]:
tot_amount2=D2.groupby('variable').agg({'raised_amount_usd':np.sum}).sort_values(ascending=False,by='raised_amount_usd')
tot_amount2

raised_amount_usd
variable                                                  
Cleantech / Semiconductors                     917829246.0
Others                                         816205523.0
News, Search and Messaging                     269140008.0
Entertainment                                  245461478.0
Manufacturing                                  237049312.0
Social, Finance, Analytics, Advertising        173673526.0
Health                                         107740187.0
Automotive & Sports                             29352177.0

In [30]:
company_inv2=D2.groupby(['variable','Company_Name'])['Company_Name'].agg(np.size).sort_values(ascending=False)
company_inv2

variable                    Company_Name  
Others                      electric-cloud    4
                            enigmatic         3
Cleantech / Semiconductors  eusa-pharma       3
Manufacturing               antenova          3
News, Search and Messaging  ubiquisys         3
                                             ..
                            wecomm            1
                            spinvox           1
                            snaptu            1
                            red-m-group       1
Automotive & Sports         prodrive          1
Name: Company_Name, Length: 264, dtype: int64

## For Country- CAN

In [31]:
con2=invest[(invest['country_code']=='CAN')&(invest['funding_round_type']=='venture')]
D3=pd.merge(con2,mapping_category)
D3

permalink  \
0                         /Organization/2Catalyze   
1                 /Organization/360Incentives-Com   
2                 /Organization/Beyond-Compliance   
3                /Organization/Camilion-Solutions   
4    /Organization/Canadian-Digital-Media-Network   
..                                            ...   
213         /Organization/Southgobi-Resources-Ltd   
214               /Organization/Sunselect-Produce   
215         /Organization/Topigen-Pharmaceuticals   
216                          /Organization/Vogogo   
217                          /Organization/Vogogo   

                               name                      homepage_url  \
0                         2Catalyze                               NaN   
1                 360incentives.com      http://www.360incentives.com   
2                 Beyond Compliance   http://www.beyondcompliance.com   
3                Camilion Solutions  http://www.support.camilion.com/   
4    Canadian Digital Media Network                    http://cdmn.ca   
..                              ...                               ...   
213             SouthGobi Resources              http://southgobi.com   
214               SunSelect Produce               http://sunselect.ca   
215         Topigen Pharmaceuticals           http://www.topigen.com/   
216                          vogogo             http://www.vogogo.com   
217                          vogogo             http://www.vogogo.com   

           category_list     status country_code state_code      region  \
0               Software  operating          CAN         ON     Toronto   
1               Software  operating          CAN         ON     Toronto   
2               Software  operating          CAN         AB     Calgary   
3               Software  operating          CAN         ON     Toronto   
4               Software  operating          CAN         ON     Toronto   
..                   ...        ...          ...        ...         ...   
213  Mining Technologies  operating          CAN         BC   Vancouver   
214          Hospitality  operating          CAN         BC  BC - Other   
215      Pharmaceuticals   acquired          CAN         QC    Montreal   
216          Curated Web        ipo          CAN         AB     Calgary   
217          Curated Web        ipo          CAN         AB     Calgary   

           city  founded_at                    Company_Name  \
0       Toronto  01-01-2008                       2catalyze   
1        Whitby  01-01-2008               360incentives-com   
2       Calgary  01-01-2000               beyond-compliance   
3       Markham  01-01-2000              camilion-solutions   
4       Toronto  01-01-2009  canadian-digital-media-network   
..          ...         ...                             ...   
213   Vancouver         NaN         southgobi-resources-ltd   
214  Aldergrove  01-06-1985               sunselect-produce   
215   Montréal         NaN         topigen-pharmaceuticals   
216     Calgary  24-01-2008                          vogogo   
217     Calgary  24-01-2008                          vogogo   

                                company_permalink  \
0                         /organization/2catalyze   
1                 /organization/360incentives-com   
2                 /ORGANIZATION/BEYOND-COMPLIANCE   
3                /organization/camilion-solutions   
4    /organization/canadian-digital-media-network   
..                                            ...   
213         /ORGANIZATION/SOUTHGOBI-RESOURCES-LTD   
214               /organization/sunselect-produce   
215         /organization/topigen-pharmaceuticals   
216                          /organization/vogogo   
217                          /ORGANIZATION/VOGOGO   

                             funding_round_permalink funding_round_type  \
0    /funding-round/d26e31e33cf217e2ed888eafc54febaf            venture   
1    /funding-round/8712b14c58f7066491d136673931ed3e            venture   
2    /

In [32]:
tot_investment3=D3.groupby('variable')['raised_amount_usd'].agg(np.size).sort_values(ascending=False)
tot_investment3

variable
Cleantech / Semiconductors                 82.0
Others                                     81.0
News, Search and Messaging                 17.0
Manufacturing                              16.0
Health                                     11.0
Social, Finance, Analytics, Advertising     8.0
Entertainment                               3.0
Name: raised_amount_usd, dtype: float64

In [33]:
tot_amount3=D3.groupby('variable').agg({'raised_amount_usd':np.sum}).sort_values(ascending=False,by='raised_amount_usd')
tot_amount3

raised_amount_usd
variable                                                  
Cleantech / Semiconductors                     755032123.0
Others                                         701944621.0
News, Search and Messaging                     147119096.0
Manufacturing                                  124582240.0
Health                                          93089714.0
Social, Finance, Analytics, Advertising         77430000.0
Entertainment                                   23110000.0

In [34]:
company_inv3=D3.groupby(['variable','Company_Name'])['Company_Name'].agg(np.size).sort_values(ascending=False)
company_inv3

variable                    Company_Name                  
Cleantech / Semiconductors  ostara                            4
                            fresco-microchip                  4
                            wavesat                           3
                            engene                            3
                            enbala-power-networks             3
                                                             ..
Others                      canadian-digital-media-network    1
                            camilion-solutions                1
                            builddirect                       1
                            blueprint-software-systems        1
Cleantech / Semiconductors  alethia-biotherapeutics           1
Name: Company_Name, Length: 173, dtype: int64

###  1. Total number of investments 
- Country-1: USA:4996
- Country-2: GBR:318
- Country-3: CAN:218

### 2.Total amount of investment 
- Country-1: USA:
- Country-2: GBR:
- Country-3: CAN:

###  3. Top sector (based on count of investments)
- Country-1: USA:-Others
- Country-2: GBR:-Cleantech/Semiconductors
- Country-3: CAN:-Cleantech/Semiconductors
 
###  4. Second-best sector (based on count of investments)
- Country-1: USA:-Cleantech/Semiconductors
- Country-2: GBR:-Others
- Country-3: CAN:-Others
 
### 5. Third-best sector (based on count of investments)
- Country-1: USA:-Health
- Country-2: GBR:-News, Search and Messaging
- Country-3: CAN:-News, Search and Messaging
 
 ### 6.  Number of investments in the top sector 
- Country-1: USA:1728
- Country-2: GBR:104
- Country-3: CAN:82

### 7.  Number of investments in the second-best sector
- Country-1: USA:1494
- Country-2: GBR:95
- Country-3: CAN:81

### 8.  Number of investments in the third-best sector 
- Country-1: USA:498
- Country-2: GBR:30
- Country-3: CAN:17

### 9.  For the top sector count-wise (point 3), which company received the highest investment?
- Country-1: USA:-Others: Company:-integrien
- Country-2: GBR:-Cleantech/Semiconductors: Company:- eusa-pharma
- Country-3: CAN:-Cleantech/Semiconductors: Company:- ostara
 
### 10.  For the second-best sector count-wise (point 4), which company received the highest investment?
- Country-1: USA:-Cleantech/Semiconductors: Company:- tigo-energy
- Country-2: GBR:-Others: Company:- electric-cloud
- Country-3: CAN:-Others: COmpany:- newstep